README.md

```diff
@@ -771,28 +777,85 @@ let numbers = create([Number], [1, 2, 3, 4]);
 
-You should be able to use to any implementation of Observables that supports `Observer.from` using [symbol-observable](https://github.com/benlesh/symbol-observable). We'll use `RxJS` for our example.
+Microstates has a mechanism to working with Microtates in reactive environments. It is called Identity. When you create an Identity from a Microstate, you get an object that has the same shape the original microstate. Every composed microstate becomes an identity and every transition gets wrapped in side effect emitting behaviour specific to that identity constructor. Let’s look at the features of the Identity and how they benefit reactive environments.
+
+## Structural Sharing
+
+A common performance optimization used by all reactive engines is to prevent re-renders for components who’s props have not changed. The most efficient way to determine if a value has not changed it to perform an exact equality check, for example: `prevValue === currentValue`. If the reference is the same, then consider the value unchanged. The Identity makes this possible with Microstates by internally managing how the Identity is constructed as a result of a transition. It will automatically determine which branches of microstates are unchanged and reuse previous identities for those branches.
+
+## Memoized Getters
+
+Microstates are immutable which makes it safe for us to memoize computations that are derived off their state. Identies will automatically memoize getters and return previously computed value when the microstate backing the identity has not changed. When a transition is invoked on the identity, the part of the identity tree that are changed will be re-created effectively invalidating the cache for the changed parts of the identity. The getters will be recomputed for state that is changed.
+
+## Debounce no-op transitions
+
+Identities automatically prevent unnecessary re-renders by debouncing transitions that do not change the value. This eliminates the need for `shouldComponentUpdate` hooks for pure components because it is safe to assume that a component that is re-rendering is re-rendering as a result of a transition that changed the value.
+
+## Identity Constructors
+
+Microstates comes with two Identity constructors: _Store_ and _Observable_. Observable will create a stream of identities. Next identity will be sent to the observer when an transition on an identity is called.
+
+### Store(microstate, callback)
+
+Store identity constructor takes two arguments: microstate and a callback. It returns an identity. When a transition is invoked on the identity, the callback will receive the next identity.
+
+```js
+import { Store, from, valueOf } from 'microstates';
+
+let initial = create(Number, 42);
+
+let last;
+
+last = Store(initial, next => (last = next));
```

Comment:
Does the `Store` callback get invoked for the initialize transition? Or just every future transition?

Reply:
To answer my own question, it seems like it's only for future transitions, so you need to use the return value from `Store` for the initial state of the `last` variable. The following _does not_ work:

```
let last;

Store(initial, next => (last = next));

last.foo.set('bax');
```

Reply:
Yes, initialize transition lazily sets up the microstate on create. When you construct the Microstate, the value is not initialized until you still pulling values off the microstate. If your microstate is bound to the DOM, the render will pull the values off the microstate which will invoke initialize but it's a pure operation without side effects.